src/EditorFeatures/Core/Implementation/InlineRename/IEditorInlineRenameService.cs

```diff
@@ -34,6 +35,32 @@ internal enum InlineRenameReplacementKind
         Complexified,
     }
 
+    internal enum InlineRenameFileRenameInfo
+    {
+        /// <summary>
+        /// This operation is not allowed
+        /// on the symbol being renamed
+        /// </summary>
+        NotAllowed,
+
+        /// <summary>
+        /// The type being renamed has multiple definition
+        /// locations which is not supported.
+        /// </summary>
+        TypeWithMultipleLocations,
+
+        /// <summary>
+        /// The type being renamed doesn't match the file
+        /// name prior to renaming
+        /// </summary>
+        TypeDoesntMatchFileName,
+
+        /// <summary>
+        /// File rename is allowed
+        /// </summary>
+        Allowed
```

Comment:
Thats true of any value ordering right? Just seems like enum values should always be added to the end. The only clever way I can think to be more explicit about that is assigning them int values.

Reply:
The public API checker for public APIs at least does reflect the actual values to prevent mistakes like this; I'm just calling out a possible warning if you did add more in the middle and F# has adopted this.